src/bindings/python/src/pyopenvino/graph/node_output.hpp

```diff
@@ -25,8 +26,20 @@ void regclass_graph_Output(py::module m, std::string typestring)
                                                                        py::dynamic_attr());
     output.doc() = docs;
 
+    // operator overloading
+    output.def(py::self < py::self);
+    output.def(py::self <= py::self);
+    output.def(py::self > py::self);
+    output.def(py::self >= py::self);
+    output.def(py::self == py::self);
+    output.def(py::self != py::self);
+
+    output.def("__hash__", [](ov::Output<VT>& port) {
+        return std::hash<VT*>()(port.get_node()) + port.get_index();
```

Comment:
@ilyachur is it valid? I remember you also implemented hash for node

Reply:
No, Node uses default python hash. I will check, it also should be implemented for node if we get nodes by value somewhere.

Reply:
We always get Nodes by shared_ptr from c++ api, so default hash operator can be used. Added a test to check that node and node.output(0) hashes are not equal.

Reply:
I have different idea in order to avoid dependencies on ptr.
Here we have a WA for python.

Reply:
@ilyachur I can't deduce approval for it. Is this WA valid and should be Python-based implementation of hash?

Reply:
In general I think in future this behavior will be changed, but at the current moment it LGTM to me